filebeat/crawler/crawler.go

```diff
@@ -42,7 +42,10 @@ func (c *Crawler) Start(prospectorConfigs []config.ProspectorConfig, eventChan c
 		logp.Debug("prospector", "File Configs: %v", prospectorConfig.Paths)
 
 		prospector, err := NewProspector(prospectorConfig, c.Registrar, eventChan)
-		c.prospectors = append(c.prospectors, prospector)
+
+		if prospector != nil {
```

Comment:
Instead of introducing the if clause, it would probably best to move this line after the error check.